pydis_site/apps/home/urls.py

```diff
@@ -1,14 +1,36 @@
+from allauth.account.views import LogoutView
+from allauth.socialaccount.views import ConnectionsView
 from django.conf import settings
 from django.conf.urls.static import static
 from django.contrib import admin
+from django.contrib.messages import ERROR
 from django.urls import include, path
 
+from pydis_site.utils.views import MessageRedirectView
 from .views import HomeView
 
 app_name = 'home'
 urlpatterns = [
     path('', HomeView.as_view(), name='home'),
     path('pages/', include('wiki.urls')),
+
+    path('accounts/', include('allauth.socialaccount.providers.discord.urls')),
+
+    path(
+        'accounts/login/cancelled', MessageRedirectView.as_view(
+            pattern_name="home", message="Login cancelled."
+        ), name='socialaccount_login_cancelled'
+    ),
+    path(
+        'accounts/login/error', MessageRedirectView.as_view(
+            pattern_name="home", message="Login failed due to an error, please try again.",
```

Comment:
"Failed due to an error" is a bit of a pleonasm. Maybe "The Login process encountered an error" or something?